design/one-pager-stacks-security-isolation.md

```diff
@@ -241,13 +235,134 @@ permissionScope: Cluster
 
 ### Stack Install
 
-Since cluster stacks and namespace stacks have differing scopes of execution in the control plane, it will be wise to allow differing access to request their installation.
-A simple way to approach this would be to create different installation request CRDs for each and then use RBAC to restrict access to them by different roles.
+Since cluster stacks and namespace stacks have differing scopes of execution in the control plane, it will be wise to
+ allow differing access to request their installation. A simple way to approach this would be to create different
+ installation request CRDs for each and then use RBAC to restrict access to them by different roles.
+
 Currently, we use a single `StackRequest` type, but going forward we should differentiate by stack type:
 
-* `ClusterStackInstall`: requests the installation of a cluster stack
-* `StackInstall`: requests the installation of a namespace stack
+* `ClusterStackInstall`: requests the installation of a cluster stack who's resources are all cluster scoped
+* `StackInstall`: requests the installation of a namespace stack who's resources are all namespace scoped
```

Comment:
> who's resource are all namespace scoped

So the Stacks Manager would only need to create / update `Role` definitions for namespace (workspace) admin, editor, viewer roles - that could be used by a client to assign permissions to
 * workspace admin role: install / upgrade app stacks via StackInstall within a namespace (workspace) + below
 * workspace editor role: manage CRs for installed app stacks -- separate from the `Role` granting access to create resource claims in the namespace (workspace)
 * workspace viewer role: view CRs for installed app stacks -- separate from the `Role` granting access to view resource claims, and the `ClusterRole` granting access to view the cluster-scoped managed resource instance

Reply:
Cluster role aggregation doesn’t exist for just roles. It may be operationally simpler to model them all as cluster roles, especially since binding a clusterRole with a rolebinding creates the same effect as if it was in a role.

If we do roles we would have to manage them without aggregation tooling from kubernetes.